Update Skeleton Instructions for Development Guide Project Setup (#97)
cisagov/development-guide: project_setup/README.md

We recommend you follow the directions below and use a skeleton for
all new repositories.

> The [`gh-skeleton`](https://github.com/cisagov/gh-skeleton) GitHub CLI

Reviewer: It looks like we link out to the `gh-skeleton` repository in multiple places. Please switch this to a reference-style link and update all uses of "gh-skeleton" to both use this link when appropriate and be formatted consistently.

Author: Hi @mcdonnnj , update to reference-style link. Addressed at 077cfbf

> tool provides the ability to easily start new projects from the

Reviewer 2: ```suggestion
GitHub CLI (`gh`) extension provides the ability to easily start new projects from the
```

## Contents ##

- [Adding a skeleton configuration to an existing repository](#adding-a-skeleton-configuration-to-an-existing-repository)

> - [Using the gh-skeleton tool to start a new repository 💀🛠](#using-the-gh-skeleton-tool-to-start-a-new-repository-)

Reviewer 2: ```suggestion
- [Using the gh-skeleton `gh` extension to start a new repository 💀🛠](#using-the-gh-skeleton-gh-extension-to-start-a-new-repository-)
```

> - [Using the gh-skeleton `gh` extension to start a new repository 💀🛠](#using-the-gh-skeleton-gh-extension-to-start-a-new-repository-)

Reviewer: The emoji no longer make sense.

```suggestion
- [Using the gh-skeleton `gh` extension to start a new repository](#using-the-gh-skeleton-gh-extension-to-start-a-new-repository-)
```

Author: Hi @mcdonnnj , removed skeleton tool emoji. Addressed at ffbea4a
https://github.com/cisagov/development-guide/pull/97/commits/ffbea4a5f404ec7840245a4eb2f9cb9edbb613b0

  - [Ansible requirement file generation tool](#ansible-requirement-file-generation-tool)
  - [Terraform IAM credentials to GitHub secrets](#terraform-iam-credentials-to-github-secrets)
  - [Managing SSM parameters from files](#managing-ssm-parameters-from-files)

## Adding a skeleton configuration to an existing repository ##

To skeletonize an existing repository, please see the guide to
[skeletonize an existing repository](skeletonize-existing-repository.md).

## Using the gh-skeleton `gh` extension to start a new repository ##

We recommend that you start your project from one of the
[skeleton projects](https://github.com/search?q=org%3Acisagov+topic%3Askeleton)
in this organization.  The [`gh-skeleton`][gh-skeleton]
`gh` extension provides the ability to easily start new projects from the existing
library of skeleton repositories.

### Selecting a skeleton ###

First, identify a suitable skeleton project to use as the starting point
for your new repository. For a list of available skeletons, see the
[Skeleton List](skeleton-list.md) or use the following command:

```console
$ gh skeleton list

> 

Reviewer 3: As long as we are updating this section, we should also update to show the current correct output:

```console
$ gh skeleton list
Available skeletons in cisagov:

skeleton-ansible-role    	A skeleton project for quickly getting a new cisagov Ansible role started.
skeleton-ansible-role-with-test-user	A skeleton project for quickly getting a new cisagov Ansible role started when that role requires an AWS test user.
skeleton-aws-lambda-python	A skeleton project for quickly getting a new Python-based AWS Lambda project started.
skeleton-docker          	A skeleton project for quickly getting a new cisagov Docker container started.
skeleton-generic         	A generic skeleton project for quickly getting a new cisagov project started.
skeleton-packer          	A skeleton project for quickly getting a new cisagov packer project started.
skeleton-python-library  	A skeleton project for quickly getting a new cisagov Python library started.
skeleton-tf-module       	A skeleton project for quickly getting a new cisagov Terraform module started.
```

Author: Hi @dav3r, update to output for gh skeleton list. Addressed at 3443b94

Reviewer 3: @dv4harr10 Why did you replace the whitespace between the skeleton repo names and their descriptions (in my sample output above) with newlines in https://github.com/cisagov/development-guide/commit/3443b94a4c2e3967ecd5407277d69abf1578accb?  Our formatting inside the `console` block should look the same as the output that users will see in their terminal.

Author: Hi @dav3r, the whitespace was giving a linter error. I can try to include it back.

Reviewer 3: Can you please show us exactly what error message you were getting?

Author: Hi @dav3r , it's resolved with 2be365c

Author: The error message was trim trailing whitespace, but it's now resolved.

Reviewer 3: Unfortunately, https://github.com/cisagov/development-guide/commit/2be365cadb2870773571224402ed0f7ae20c2320 still does not address my original concern:

> Why did you replace the whitespace between the skeleton repo names and their descriptions (in my sample output above) with newlines in https://github.com/cisagov/development-guide/commit/3443b94a4c2e3967ecd5407277d69abf1578accb?

Author: Hi @dav3r , please give me a call, I posted my number to the chat Thanks

Author: Hi @dav3r, update to output for gh skeleton list should be resolved at 8513d28

Reviewer 3: @dv4harr10 Looks good now with https://github.com/cisagov/development-guide/pull/97/commits/8513d287209b2f0c4ab81d68af244759eea753b9, thanks!

To skeletonize an existing repository, please see the guide to
[skeletonize an existing repository](skeletonize-existing-repository.md).

> ## Using the gh-skeleton tool to start a new repository 💀🛠 ##

Reviewer 2: ```suggestion
## Using the gh-skeleton `gh` extension to start a new repository 💀🛠 ##
```

We recommend that you start your project from one of the
[skeleton projects](https://github.com/search?q=org%3Acisagov+topic%3Askeleton)
in this organization.  The [`gh-skeleton`](https://github.com/cisagov/gh-skeleton)

> tool provides the ability to easily start new projects from the existing

Reviewer 2: ```suggestion
`gh` extension provides the ability to easily start new projects from the existing
```

[Skeleton List](skeleton-list.md) or use the following command:

```bash
gh skeleton list
```

> ```console

Reviewer: Please combine these codeblocks and ensure the sample output aligns with the output of the extension. Please also update the lead-in formatting like so:

```
```console
$ gh skeleton list
```

Author: Hi @mcdonnnj , updated, replace text at line 54 to 58 Readme.md. Addressed at 
https://github.com/cisagov/development-guide/pull/97/commits/ffbea4a5f404ec7840245a4eb2f9cb9edbb613b0

### Cloning a selected skeleton ###

Next, use the [`gh-skeleton`](https://github.com/cisagov/gh-skeleton)

> tool to clone, rename, and prepare the contents of your new repository

Reviewer 2: ```suggestion
`gh` extension to clone, rename, and prepare the contents of your new repository
```

for publication.  The tool will print out each command it is issuing
and its result.

```console

> gh skeleton clone [options] <parent-repo-name> <new-repo-name>

Reviewer: ```suggestion
$ gh skeleton clone [options] <parent-repo-name> <new-repo-name>
```

Author: Hi @mcdonnnj , I received 2 linter failed errors for adding the dollar sign prompt, it complains that expected output is required. I worked with Michael on this solution to add the following: 
```console
$ gh skeleton clone [options] <parent-repo-name> <new-repo-name>
# Expected output:
# Cloning into 'new-repo-name' .  
```
Alternatively we may need to omit the dollar sign prompt. What do you suggest?

Author: Hi @mcdonnnj , the change above satisfied the linter. Addressed at 077cfbf

```

For example, to create a project based on `skeleton-ansible-role` named
`ansible-role-quantum-rng` in your local `~/projects` directory:

> ```bash

Reviewer: ```suggestion
```console
```

Author: Hi @mcdonnnj , updated text line 100. Addressed at 
https://github.com/cisagov/development-guide/pull/97/commits/ffbea4a5f404ec7840245a4eb2f9cb9edbb613b0

> gh skeleton clone --change-dir ~/projects skeleton-ansible-role ansible-role-quantum-rng

Reviewer: ```suggestion
$ gh skeleton clone --change-dir ~/projects skeleton-ansible-role ansible-role-quantum-rng
```

## Create and publish the GitHub repository ##

> Once the `gh-skeleton` tool has run, the next step is to publish to a GitHub

Reviewer 2: ```suggestion
Once the `gh-skeleton` `gh` extension has run, the next step is to publish to a GitHub
```
